source/kernel/src/lib.rs

```diff
@@ -294,43 +295,70 @@ impl Kernel {
+                    .expect("failed to spawn SerialMuxService initialization"),
+            )
+        } else {
+            None
+        };
 
         // Initialize the Forth spawnulator.
-        self.initialize(SpawnulatorServer::register(self, settings.spawnulator))
-            .expect("failed to spawn SpawnulatorService initialization");
+        if let Some(spawnulator) = settings.spawnulator {
+            self.initialize(SpawnulatorServer::register(self, spawnulator))
+                .expect("failed to spawn SpawnulatorService initialization");
+        }
 
         // Initialize Serial Mux daemons.
-        self.initialize(async move {
-            sermux_up
-                .await
-                .expect("SerialMuxService initialization should not be cancelled")
-                .expect("SerialMuxService initialization failed");
-
-            #[cfg(feature = "serial-trace")]
-            crate::serial_trace::SerialSubscriber::start(self, settings.sermux_trace).await;
-
-            self.spawn(daemons::sermux::loopback(self, settings.sermux_loopback))
-                .await;
-            tracing::debug!("SerMux loopback started");
-
-            self.spawn(daemons::sermux::hello(self, settings.sermux_hello))
-                .await;
-            tracing::debug!("SerMux Hello World started");
-        })
-        .expect("failed to spawn default serial mux service initialization");
+        if let Some(sermux_up) = sermux_up {
+            self.initialize(async move {
+                sermux_up
+                    .await
+                    .expect("SerialMuxService initialization should not be cancelled")
+                    .expect("SerialMuxService initialization failed");
+
+                #[cfg(feature = "serial-trace")]
+                if let Some(sermux_trace) = settings.sermux_trace {
+                    crate::serial_trace::SerialSubscriber::start(self, sermux_trace).await;
+                }
+
+                if let Some(sermux_loopback) = settings.sermux_loopback {
+                    self.spawn(daemons::sermux::loopback(self, sermux_loopback))
+                        .await;
+                    tracing::debug!("SerMux loopback started");
+                }
+
+                if let Some(sermux_hello) = settings.sermux_hello {
+                    self.spawn(daemons::sermux::hello(self, sermux_hello)).await;
+                    tracing::debug!("SerMux Hello World started");
+                }
+            })
+            .expect("failed to spawn default serial mux service initialization");
+        } else {
+            let deps = [
+                #[cfg(feature = "serial-trace")]
+                settings.sermux_trace.is_some(),
+                settings.sermux_loopback.is_some(),
+                settings.sermux_hello.is_some(),
+            ];
+
+            if deps.into_iter().any(identity) {
+                tracing::error!("Sermux services configured without sermux! Skipping.");
```

Comment:
if sermux is disabled, this trace...probably won't go anywhere, unless an alternate tracing sink is enabled...

Reply:
Yeah, it was "trace or panic", and I went with trace. It was a coin toss tho, so if you prefer panic we can do that.